packages/docusaurus/src/client/RouteAnnouncer.tsx

```diff
@@ -0,0 +1,78 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import React from 'react';
+import {Portal} from './exports/Portal';
+
+type Props = {
+  // Force an update on route transition
+  // eslint-disable-next-line react/no-unused-prop-types
+  location: string;
+  children: React.ReactNode;
+};
+
+type State = {
+  routeAnnouncement: string;
+};
+
+class RouteAnnouncerWrapper extends React.Component<Props, State> {
+  constructor(props: Props) {
+    super(props);
+    this.state = {
+      routeAnnouncement: '',
+    };
+  }
+
+  override componentDidUpdate(): void {
+    const {routeAnnouncement} = this.state;
+
+    requestAnimationFrame(() => {
+      const firstHeading = document.querySelectorAll(`#__docusaurus h1`)[0];
```

Comment:
this is a theme-related selector, so it should rather be in the theme

(although it's unlikely 2 themes will have different selectors 🤷‍♂️ )